pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1423,7 +1423,16 @@ func (l *LibvirtDomainManager) getDomainSpec(dom cli.VirDomain) (*api.DomainSpec
 	if err != nil {
 		return nil, err
 	}
-	return util.GetDomainSpec(state, dom)
+
+	domainSpec, err := util.GetDomainSpecWithRuntimeInfo(state, dom)
+	if err != nil {
+		// Return without runtime info only for cases we know for sure it's not supposed to be there
+		if domainerrors.IsNotFound(err) || domainerrors.IsInvalidOperation(err) {
+			return util.GetDomainSpec(state, dom)
+		}
```

Comment:
how did you determine these are the errors to check for?

Reply:
that's what we do in https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-launcher/notify-client/client.go#L229